modules/extensions/main/process-center/extensions/app/greg-publisher-defaults/themes/default/js/notes-api.js

```diff
@@ -0,0 +1,168 @@
+                        renderPartial('notes-note-container',input,function(result){
+                            $('#collapseNotes .wr-panel-notes').append(result);
+                        });
+                    });
+                    $('#add-note-content').val('');
+                messages.alertSuccess("Note added successfully");
+                setTimeout(function(){location.reload(true);},2000);
+                location.reload(true);
+            },
+            error: function () {
+                console.log("Error adding a note.");
+            }
+        });
+    });
+
+    /**
+     * This function is used to get reply container Id.
+     *
+     * @param id    id of the root note.
+     * @returns {string}
+     */
+    var replyContainerId = function (id) {
+        return '#' + id + 'reply';
+    };
+
+    /**
+     * This function is used to get reply notes.
+     */
+    $('.reply-note').on('click', function () {
+        var id = $(this).data('id');
+        var noteContainer = this;
+        var data = {};
+        data.overview_resourcepath = this.dataset.path;
+        data.overview_note = $(replyContainerId(id)).val();
+        data.overview_replypath = this.dataset.path;
+        data.overview_visibility = "public";
+        data.overview_status = "Open";
+
+        $.ajax({
+            url: caramel.context + "/apis/assets?type=note",
+            type: 'POST',
+            data: JSON.stringify(data),
+            contentType: "application/json",
+            dataType: 'json',
+            success: function (response) {
+                /*renderPartial('notes-note',response.list,function(result){
+                    $(noteContainer).closest('div.well').siblings('.wr-panel-sub-note').append(result);
+                });*/
+                $(replyContainerId(id)).val('');
+                messages.alertSuccess("Reply added successfully");
+                setTimeout(function(){location.reload(true);},2000);
+            },
+            error: function () {
+                console.log("Error while adding reply");
+            }
+        });
+    });
+
+    /**
+     * This function is used to get reply notes.
+     */
+    $('.wr-panel-note > .wr-panel-msg').on('click', function() {
+
+        var path = $(this).closest('.wr-panel-note').data('path');
+        var id = $(this).closest('.wr-panel-note').attr('href');
+
+        $.ajax({
+            url: caramel.context + '/apis/assets?type=note&q="overview_replypath":"' + path + '"',
+            type: 'GET',
+            success: function (response) {
+                $(id + "> .wr-panel-sub-note").html("");
+                $.each(response.list, function(key, value){
+                    renderPartial('notes-note',value,function(result){
+                        $(id + "> .wr-panel-sub-note").append(result);
+                    });
+                });
+            },
+            error: function () {
+                console.log("Error loading note replies.");
```

Comment:
Console log needs to remove or else need to check console availability. this will break the code on IE